src/Microsoft.Identity.Client/Internal/Requests/RequestBase.cs

```diff
@@ -235,6 +240,29 @@ protected AuthenticationResult CacheTokenResponseAndCreateAuthenticationResult(M
             }
         }
 
+        internal void CheckResponseFromBroker(MsalTokenResponse msalTokenResponse)
+        {                
+            AuthenticationRequestParameters.RequestContext.Logger.Info(LogMessages.CheckMsalTokenResponseReturnedFromBroker);
+            if(msalTokenResponse.AccessToken != null)
+            {
+                AuthenticationRequestParameters.RequestContext.Logger.Info(
+                    LogMessages.BrokerResponseContainsAccessToken + 
+                    msalTokenResponse.AccessToken.Count());
+                return;
+            }
+            else if(msalTokenResponse.Error != null)
+            {
+                AuthenticationRequestParameters.RequestContext.Logger.Info(
+                    LogMessages.ErrorReturnedInBrokerResponse(msalTokenResponse.Error));
```

Comment:
>msalTokenResponse.Error) [](start = 62, length = 24)

why log and throw exceptions if the token response is doesn't contain an error ?

Reply:
we have similar logic in adal. As i mentioned, i'm not sure what scenario would result in no error returned, or we assume we'll always get an error back? if so, then this can be modified. i'll check w/jason on this.

---
In reply to: [256613705](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/pull/881#discussion_r256613705) [](ancestors = 256613705)